Feat/30661 start pi cancel by camunda client

stoerti opened the pull request:
## Description

This adds the first batchOperation creation command to the Camunda Client. The API currently uses two steps:

```
camundaClient
  .newCreateBatchOperationCommand()
  .processInstanceCancel()
  .filter(filter)
  .send();
```

1. The first step creates a generic CreateBatchOperationCommand
2. The second step then specializes the command to a specific type and can apply a filter
3. possible additional generic steps can be added after this until the command is sent to the REST gateway

Note, that currently only the REST gateway is supported. Currently we have no GRPS representation of any search filter available.

The REST API also introduces a new BatchOperationTypeEnum (instead if the previous String). This is used to temporarily store the type in the command builder.

## Related issues

closes #30661

p-wunderlich reviewed (commented):
LGTM! BUT, not a single test? Or will you provide them with the query client?

stoerti commented:
> BUT, not a single test? Or will you provide them with the query client?

I have a test together with the query side, but I also added specific unit tests for the command side. I've also refactored the commandImpl a bit to also support fluent filters.

mustafadagher commented on clients/java/src/main/java/io/camunda/client/impl/command/CreateBatchOperationCommandImpl.java:
```diff
@@ -0,0 +1,129 @@
+
+  @Override
+  public CreateBatchOperationCommandStep3<E> filter(final E filter) {
+    this.filter = filter;
```
🔧 should we require non-null here?

mustafadagher commented on clients/java/src/main/java/io/camunda/client/impl/command/CreateBatchOperationCommandImpl.java:
```diff
@@ -0,0 +1,129 @@
+
+  @Override
+  public CreateBatchOperationCommandStep3<E> filter(final Consumer<E> fn) {
+    filter = filterFactory.get();
```
same
